engine.py

```diff
@@ -1,113 +1,81 @@
-## This is the main chess game engine that implements the rules of the game
-## and stores the state of the the chess board, including its pieces and moves
+# This is the main chess game engine that implements the rules of the game
+# and stores the state of the the chess board, including its pieces and moves
 
 
 class Game_state():
-	
-	def __init__(self):
-		"""
-			The chess board is an 8 X 8 dimensional array (Matrix of 8 rows and 8 columns )
-			i.e a list of lists. Each element of the Matrix is a string of two characters 
-			representing the chess pieces in the order "type" + "colour"
 
-			light pawn = pl
-			dark pawn  = pd
+    def __init__(self):
+        """
+            The chess board is an 8 X 8 dimensional array (Matrix of 8 rows and 8 columns )
+            i.e a list of lists. Each element of the Matrix is a string of two characters
+            representing the chess pieces in the order "type" + "colour"
+            light pawn = pl
+            dark pawn  = pd
+            light knight = nl
+            dark knight  = nd
+            e.t.c
+            empty board square = "  " ---> double empty space
+        """
+
+        self.board = [
+
+            ["rd", "nd", "bd", "qd", "kd", "bd", "nd", "pl"],
+            ["  ", "  ", "  ", "  ", "  ", "  ", "  ", "  "],
+            ["pd", "  ", "  ", "  ", "  ", "  ", "  ", "  "],
+            ["  ", "  ", "pd", "  ", "  ", "  ", "  ", "  "],
+            ["  ", "  ", "  ", "  ", "  ", "  ", "  ", "pd"],
+            ["  ", "  ", "  ", "  ", "  ", "  ", "  ", "pl"],
+            ["  ", "  ", "  ", "  ", "  ", "  ", "  ", "  "],
+            ["rl", "nl", "pd", "ql", "pd", "bl", "pd", "rl"]]
+
+        self.light_to_move = True  # True = light's turn to play; False = dark's turn to play
+        self.move_log = []  # keeps a log of all moves made withing a game
+        self.move_piece = {"p": self.get_pawn_moves, "r": self.get_rook_moves, \
+                           "q": self.get_queen_moves, "k": self.get_king_moves, \
+                           "b": self.get_bishop_moves, "n": self.get_knight_moves}
+
+    def get_pawn_moves(self, r, c, moves):
+        """
+            Calculates all possible pawn moves for a given color (light or dark)
+            and appends them to a list
+            input parameter(s):
+            r     --> starting row (int)
+            c     --> starting colum (int)
+            moves --> possible moves container (list)
+            return parameter(s):
+            None
+        """
```

Comment:
please run your code thoroughly before initiating a pull request to ensure minor bugs are fixed from your end

Reply:
I run the code countless times before sending it sir. I think the indentation errors happened because I wrote the code using anaconda which may be have different indent settings, but I have rewrote the code with pycharm and I think everything is working perfectly. Thanks for your time